### PR TITLE
[ROCm] Update CI to recently released ROCm 2.1 release

### PR DESCRIPTION
**Pull request by iotamudelta**
* we do not need EAP packages any longer as the antistatic feature is now in the release
* consistently install the rccl package
* Skip one unit test that has regressed with 2.1
* Follow-up PRs will use 2.1 features once deployed on CI

### CODE REVIEW

**Comment by SsnL** [comment]
nice, could you kill this too?
https://github.com/pytorch/pytorch/blob/887080e92abc16146543e60d64980c2c533cbed4/test/common_utils.py#L398-L402

**Comment by iotamudelta** [comment]
@SsnL I believe that one we will be able to remove with 2.1. If not, we'll look into solving that one.

**Comment by petrex** [comment]
@bddppq Let's coordinate rocm upgrade (roct compatibility)

**Comment by bddppq** [comment]
new docker images build https://ci.pytorch.org/jenkins/job/caffe2-docker-trigger/248/

**Comment by bddppq** [comment]
@iotamudelta @petrex Do we still need the custom Thrust in centos build? 
https://github.com/pytorch/pytorch/blob/541f277467f7f7a5885bc8f1aa25dc1861691362/docker/caffe2/jenkins/common/install_rocm.sh#L87-L91

File: docker/caffe2/jenkins/common/install_rocm.sh [hunk 1]
```diff
@@ -86,15 +78,6 @@ install_centos() {
...
-  pushd /tmp
-  wget https://github.com/RadeonOpenCompute/hcc/releases/download/roc-1.9.2-pytorch-eap/hcc-1.2.18473-Linux.rpm
-  wget https://github.com/ROCm-Developer-Tools/HIP/releases/download/roc-1.9.2-pytorch-eap/hip_base-1.5.18462.rpm
-  wget https://github.com/ROCm-Developer-Tools/HIP/releases/download/roc-1.9.2-pytorch-eap/hip_doc-1.5.18462.rpm
-  wget https://github.com/ROCm-Developer-Tools/HIP/releases/download/roc-1.9.2-pytorch-eap/hip_hcc-1.5.18462.rpm
-  wget https://github.com/ROCm-Developer-Tools/HIP/releases/download/roc-1.9.2-pytorch-eap/hip_samples-1.5.18462.rpm
-  rpm -i --replacefiles ./hcc-1.2.18473-Linux.rpm ./hip_base-1.5.18462.rpm ./hip_hcc-1.5.18462.rpm ./hip_doc-1.5.18462.rpm ./hip_samples-1.5.18462.rpm
-  popd
-  
   # Cleanup
```
Comment (bddppq) [inline docker/caffe2/jenkins/common/install_rocm.sh]: These cleanup lines can be deleted as well

**Comment by bddppq** [comment]
A lot of warnings in https://ci.pytorch.org/jenkins/job/pytorch-builds/job/py2-clang7-rocmdeb-ubuntu16.04-build/16183/console

```
warning: <unknown>:0:0: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering
```
```
/var/lib/jenkins/workspace/aten/src/THH/THHSortUtils.cuh:141:1: warning: loop not unrolled: the optimizer was unable to perform the requested transformation; the transformation might be disabled or specified as part of an unsupported transformation ordering [-Wpass-failed=transform-warning]
```

**Comment by iotamudelta** [comment]
@bddppq the warnings originate from the amdgpu backend in LLVM. They are associated with not being able to unroll a loop even though an explicit pragma requested that and are only performance warnings.

**Comment by iotamudelta** [comment]
@bddppq Concerning Thrust: yes, I'd prefer if we for now keep the checkout if that's OK. There are some bigger changes incoming w.r.t. Thrust on our side.

**Comment by bddppq** [comment]
@iotamudelta @petrex Looks like the docker image with new rocm version works fine with old version of kernel driver. Are there any known backward incompatibility? If not then we can update the CI docker image and the drivers on the amd workers separately.

**Comment by iotamudelta** [comment]
@bddppq technically running 2.1 on 1.9.2 is not supported/tested. That being said: a) I've been running it myself on my host, b) if we were to update the hosts first to 2.1, running 1.9.2 on 2.1 is also not tested/supported. So I think we can go whichever way you prefer and then, if we update the dockers first, just make sure that the hosts get updated in short order.

**Comment by petrex** [comment]
@bddppq my concern is mainly on thunk compatibility. Lets find a way to upgrade bare-meal (maybe with help from devops) and deploy the new docker.

**Comment by bddppq** [comment]
@petrex @iotamudelta Ok I agree we should upgrade the driver to 2.1, let's sync to rollout the upgrade on the workers.
I'm going to land this PR and bump the CI to use new docker images soon since from our testing they are passing.

**Pull request closed without merging**